pkg/service/restore/model.go

```diff
@@ -33,9 +34,14 @@ type Target struct {
 	RestoreSchema   bool       `json:"restore_schema,omitempty"`
 	RestoreTables   bool       `json:"restore_tables,omitempty"`
 	Continue        bool       `json:"continue"`
+	DCMappings      DCMappings `json:"dc-mapping"`
 
 	// Cache for host with access to remote location
 	locationHosts map[Location][]string `json:"-"`
+	// Cache for host and their DC after applying DCMappings
+	hostDCs map[string][]string
+	// Cache for dcs that shouldn't be restored from location
+	ignoredSourceDC []string
```

Comment:
Wouldn't it be way simpler if we just had a single mapping `map[string][string]` of dst DC -> src DC list?
It could replace all `locationHosts, hostDCs, ignoredSourceDC`.
During batch dispatch we could/should have access to given hosts DC, so it would give us enough information for correctness.
Ignored DCs could be replaced with restored DCs which could be easily obtained from such mapping when needed for filtering.

The assumption is that we don't need to care for cases like: what if only a few hosts from this DC have access to this location. Even when making backup, the whole DC needs to be backed up to a single location and all hosts from given DC needs to have access to it.